### PR TITLE
dev/core#2308 do not require fields if activity_id is present

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#2308 do not require fields if activity_id is present - towards supporting update by import

Before
----------------------------------------
requirement for activity type & date fields still enforced when id is present

After
----------------------------------------
Not enforced, in this place

Technical Details
----------------------------------------
Note this is only one place they are required & it will still fail later with this merged

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Activity/Import/Parser/Activity.php [hunk 1]
```diff
@@ -132,9 +132,21 @@ public function summary(&$values) {
...
 
     try {
+      // Check required fields if this is not an update.
+      if (!$this->getFieldValue($values, 'activity_id')) {
+        if (!$this->getFieldValue($values, 'activity_label')
+        && !$this->getFieldValue($values, 'activity_type_id')) {
+          throw new CRM_Core_Exception(ts('Missing required fields: Activity type label or Activity type ID'));
+        }
+        if (!$this->getFieldValue($values, 'activity_date_time')) {
+          throw new CRM_Core_Exception(ts('Missing required fields'));
```
Comment (seamuslee001) [inline CRM/Activity/Import/Parser/Activity.php:142]: @eileenmcnaughton any reason why you split the activity_date_time out from the other 2?

Comment (eileenmcnaughton) [reply CRM/Activity/Import/Parser/Activity.php:142]: The other 2 are an either or so a bit more complex

File: CRM/Activity/Import/Parser/Activity.php [hunk 2]
```diff
@@ -147,23 +159,11 @@ public function summary(&$values) {
...
-    $dateType = $session->get('dateTypes');
     if (!isset($params['source_contact_id'])) {
       $params['source_contact_id'] = $session->get('userID');
     }
     foreach ($params as $key => $val) {
-      if ($key === 'activity_date_time') {
-        if ($val) {
-          $dateValue = CRM_Utils_Date::formatDate($val, $dateType);
-          if ($dateValue) {
-            $params[$key] = $dateValue;
```
Comment (demeritcowboy) [inline CRM/Activity/Import/Parser/Activity.php:159]: I think removing this params assignment is ok. It gets passed on to isErrorInCustomData below but this field isn't custom data. In testing it still catches invalid dates.

And just noting this is now a silly loop, looping thru all the params just to find engagement_level. And the comments on line 160 and 172 no longer apply.

**Comment by demeritcowboy** [comment]
This looks ok to me. I'm assuming "it will still fail later with this merged" is referring to the overall general concept of updating by providing an activity id, not that requirement checking for new activities is now broken (it doesn't seem to be broken).

**Comment by eileenmcnaughton** [comment]
@demeritcowboy I do actually have a folow on patch on engagement level - https://github.com/civicrm/civicrm-core/pull/19421 - I can push it into this if you want

**Comment by eileenmcnaughton** [comment]
Actually probably a separate PR since you have OK'd this - @seamuslee001 wanna merge it ?

Note the chain eventually winds up with https://github.com/civicrm/civicrm-core/pull/19421/files#diff-6347a7ef5c13763509b0cbbf696e881473ea89931a7ae2859870719a69986896R341

**Pull request merged**